src/main/java/org/jenkinsci/plugins/github_branch_source/Endpoint.java

```diff
@@ -48,19 +56,85 @@
  * @author Stephen Connolly
  */
 public class Endpoint extends AbstractDescribableImpl<Endpoint> {
+    /**
+     * Common prefixes that we should remove when inferring a display name.
+     */
+    private static final String[] COMMON_PREFIX_HOSTNAMES = {
+            "git.",
+            "gitea.",
+            "gogs.",
+            "vcs.",
+            "scm.",
+            "source."
+    };
+
     private final String name;
     private final String apiUri;
 
+    /**
+     * Makes best effort to guess a "sensible" display name from the hostname in the apiUri.
+     *
+     * @param apiUri the apiUri.
+     * @return the display name or {@code null}
+     * @throws LinkageError if Guava changes their API that we have depended on.
+     */
+    @CheckForNull
+    /*package*/ static String inferDisplayName(@CheckForNull String apiUri) throws LinkageError {
+        if (apiUri == null) {
+            return apiUri;
+        }
+        String hostName;
+        try {
+            URI serverUri = new URI(apiUri);
+            hostName = serverUri.getHost();
+            if (hostName != null) {
+                // let's see if we can make this more "friendly"
+                InternetDomainName host = InternetDomainName.from(hostName);
+                if (host.hasPublicSuffix()) {
+                    String publicName = host.publicSuffix().name();
+                    hostName = StringUtils.removeEnd(StringUtils.removeEnd(host.name(), publicName), ".")
+                            .toLowerCase(Locale.ENGLISH);
+                } else {
+                    hostName = StringUtils.removeEnd(host.name(), ".").toLowerCase(Locale.ENGLISH);
+                }
+                for (String prefix : COMMON_PREFIX_HOSTNAMES) {
+                    if (hostName.startsWith(prefix)) {
+                        hostName = hostName.substring(prefix.length());
+                        break;
+                    }
+                }
+            }
+        } catch (URISyntaxException e) {
+            // ignore, best effort
+            hostName = null;
+        }
+        return hostName;
+    }
+
     @DataBoundConstructor
     public Endpoint(String apiUri, String name) {
-        this.apiUri = Util.fixEmptyAndTrim(apiUri);
-        this.name = Util.fixEmptyAndTrim(name);
+        this.apiUri = GitHubConfiguration.normalizeApiUri(Util.fixEmptyAndTrim(apiUri));
+        name = Util.fixEmptyAndTrim(name);
+        if (name == null) {
+            this.name = inferDisplayName(apiUri);
+        } else {
+            this.name = name;
+        }
+    }
+
+    private Object readResolve() throws ObjectStreamException {
```

Comment:
Class is not private. Would readResolve() work correctly if you override the class? IIRC in XStream there were some problems with that a couple of years ago